### PR TITLE
Fix scalar check in RuntimeInfoHook

### PR DESCRIPTION
**Pull request by i-aki-y**
## Motivation

Some numpy scalars (float32, float16, ...) are misclassified as non-scalar values inRuntimeInfoHook.

See:

```
from mmengine.hooks.runtime_info_hook import _is_scalar
v = np.mean([0.5])
print(_is_scalar(v.astype(np.float16)))  # => False
print(_is_scalar(v.astype(np.float32)))  # => False
print(_is_scalar(v.astype(np.float64)))  # => True
```

This causes problems when the custom metrics are calculated in numpy because the log_processor looks to collect only scalar values as metrics to give the visualizer them.

## Modification

Added np.floating and np.integer as valid scalar types.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

I encountered this problem when I defined custom metrics where I calculated metrics in tensor and converted the result to numpy. (ex. metrics.detach().cpu().numpy() --> np.float32)

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmengine/hooks/runtime_info_hook.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def _is_scalar(value: Any) -> bool:
     """
     if isinstance(value, np.ndarray):
         return value.size == 1
-    elif isinstance(value, (int, float)):
+    elif isinstance(value, (int, float, np.floating, np.integer)):
```
Comment (zhouzaida) [inline mmengine/hooks/runtime_info_hook.py:26]: Can we replace `np.floating, np.integer` with `np.number`?

Comment (i-aki-y) [reply mmengine/hooks/runtime_info_hook.py]: @zhouzaida  Thank you for your comment. Yes, I missed it. 
The np.number would be better.

For clarification, a difference is that np.number includes complex numbers while the `np.floating, np.integer` does not.
https://numpy.org/doc/stable/reference/arrays.scalars.html

**Review by zhouzaida — APPROVED** [review]
